apps/vercel/frontend/src/clients/Vercel.ts

```diff
@@ -146,9 +156,61 @@ export default class VercelClient implements VercelAPIClient {
     return data.deployments[0];
   }
 
+  async validateProjectContentfulSpaceId(
+    currentSpaceId: string,
+    projectId: string,
+    teamId: string
+  ) {
+    try {
+      const data = await this.listProject(projectId, teamId);
+      const envs = data.env;
+      const contentfulSpaceIdEnv = envs.find((env) => env.key === CONTENTFUL_SPACE_ID);
+      if (contentfulSpaceIdEnv) {
+        const res = await fetch(
+          `${this.baseEndpoint}/v1/projects/${projectId}/env/${
+            contentfulSpaceIdEnv.id
+          }?${this.buildTeamIdQueryParam(teamId)}`,
+          {
+            headers: this.buildHeaders(),
+            method: 'GET',
+          }
+        );
+
+        const data = await res.json();
+        return data.value === currentSpaceId;
+      }
+    } catch (e) {
+      // let user continue if there is an error validating env value
+      console.error(e);
+    }
+
+    return false;
+  }
+
+  private async listProject(projectId: string, teamId: string): Promise<Project> {
```

Comment:
Super small naming thing but I tend to think of "list" as returning a collection and would suggest a better verb like `getProject` or `fetchProject`.

Reply:
great callout, have adjusted this!